runtime/src/append_vec.rs

```diff
@@ -379,36 +378,21 @@ impl<'a> serde::de::Visitor<'a> for AppendVecVisitor {
         let file_size: u64 = deserialize_from(&mut rd).map_err(Error::custom)?;
         let offset: usize = deserialize_from(&mut rd).map_err(Error::custom)?;
 
-        let split_path: Vec<&str> = path.to_str().unwrap().rsplit('/').collect();
```

Comment:
this code relies on a global variable being populated by AccountsDb::update_from_stream(), and I couldn't make it work with the new code.  @sambley if you can resurrect without a global and with a unit test, that'd be awesome.  if not, lemme know.

Reply:
@rob-solana, will look into removing the dependency on the global and adding a unit test